package/gluon-mesh-vpn-wireguard/files/lib/netifd/proto/gluon_wireguard.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/sh
+# Copyright 2016-2017 Christof Schulze <christof@christofschulze.com>
+# Licensed to the public under the Apache License 2.0.
+
+. /lib/functions.sh
+. ../netifd-proto.sh
+init_proto "$@"
+
+proto_gluon_wireguard_init_config() {
+  no_device=1
+  available=1
+  renew_handler=1
+}
+
+proto_gluon_wireguard_renew() {
+  local config="$1"
+  echo wireguard RENEW: $@
+  ifdown $config
+  ifup $config
+}
+
+proto_gluon_wireguard_setup() {
+  local config="$1"
+  ifname=$(uci get network.$config.ifname) # we need uci here because nodevice=1 means the device is not part of the ubus structure
+
+  if [[ $(wg show all latest-handshakes |wc -l) -ge $(uci get wireguard.mesh_vpn_backbone.peer_limit) ]]; then
+    echo "not establishing another connection, we already have  $(uci get wireguard.mesh_vpn_backbone.peer_limit) connections." >&2
+    ip link del $ifname
+    ifdown $config
+    exit 1
+  fi
+
+  (
+  flock -n 9
+
+  if [[ $(uci get wireguard.mesh_vpn_backbone.enabled) -eq 1 ]]; then
```

Comment:
A Gluon package should not create a config file `wireguard`. Better options would be:
- Call this file `gluon-mesh-vpn-wireguard`
- Add all configuration that is necessary to `gluon` and leave the rest out of UCI altogether (I think only the private key is actually required, and the rest could be taken from site.conf directly?)

Reply:
when we have peer groups we need to store configuration. I am happy to rename this to gluon-mesh-vpn-wireguard.

Reply:
VPN enable/disable state is already managed in `/etc/config/gluon` independently of used VPN software.

Reply:
the netifd protocols are implemented in shell. dynamically creating this content and thus migrating the logic from the upgrade script into the protocol implementation implies migrating the parsing of site.conf into the protocol. I agree most of the config is contained in the site.conf but this requires some other mechanism.

I could parse this in an upgrade script into a shell file to be sourced that defines variables and an array for the peers and go from there but if this is really an improvement over  the existing implementation is doubtful. Considering that the domain evaluation has to be completed to successfully implement the peer handling even though I like the idea of directly taking the data from site, I do not think this is feasible in this very case. Maybe after we have built some more tooling?

I did rename the uci file.

Reply:
And I started using /etc/config/gluon for enabling/disabling vpn.

Reply:
nevermind the ramblings about the domain stuff - I am using gluon-show-site and got this implemented